src/AvaloniaMath/Rendering/AvaloniaElementRenderer.cs

```diff
@@ -49,16 +52,17 @@ public void RenderGlyphRun(Func<double, GlyphRun> scaledGlyphFactory, double x,
 
                 // TODO fix baseline text display
                 var origin = glyphRun.Position.WithY(glyphRun.Position.Y-tf.FontSize*0.75);
-                
-                _drawingContext.DrawText(foreground, origin, ft);
-                
+
+                _drawingContext.DrawText(brush, origin, ft);
             }
         }
 
-        public void RenderRectangle(Rect rectangle, IBrush foreground)
+        public void RenderRectangle(Rect rectangle, RgbaColor? foregroundColor)
         {
+            // TODO: foregroundColor can be null?
+            var brush = new SolidColorBrush(foregroundColor.Value.ToAvaloniaColor());
```

Comment:
Does it need nullable or non-nullable here?

Reply:
Maybe: ```foregroundColor.HasValue ? foregroundColor.Value.ToAvaloniaColor() : null```

Reply:
I would make it non-nullable, until we find a case when `null` is useful.

Reply:
We can forward null to this function and only here make a comparison and get Color.Black from wpf / avalonia